built_value/test/standard_json_plugin_test.dart

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2016, Google Inc. Please see the AUTHORS file for details.
+// All rights reserved. Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/standard_json_plugin.dart';
+import 'package:built_value/serializer.dart';
+import 'package:test/test.dart';
+
+void main() {
+  final serializers = (new Serializers().toBuilder()
+        ..addPlugin(new StandardJsonPlugin())
+        ..addBuilderFactory(
+            const FullType(BuiltList, const [const FullType(int)]),
+            () => new ListBuilder<int>())
+        ..addBuilderFactory(
+            const FullType(BuiltList, const [
+              const FullType(BuiltList, const [const FullType(int)])
+            ]),
+            () => new ListBuilder<BuiltList<int>>())
+        ..addBuilderFactory(
+            const FullType(
+                BuiltMap, const [const FullType(int), const FullType(String)]),
+            () => new MapBuilder<int, String>())
+        ..addBuilderFactory(
+            const FullType(BuiltMap,
+                const [const FullType(String), const FullType(String)]),
+            () => new MapBuilder<String, String>())
+        ..addBuilderFactory(
+            const FullType(BuiltMap, const [
+              const FullType(BuiltMap,
+                  const [const FullType(int), const FullType(String)]),
+              const FullType(
+                  BuiltMap, const [const FullType(int), const FullType(String)])
+            ]),
+            () =>
+                new MapBuilder<BuiltMap<int, String>, BuiltMap<int, String>>()))
+      .build();
+
+  group('Serializers with StandardJsonPlugin', () {
+    test('throws on serialize if specifiedType is missing', () {
+      expect(() => serializers.serialize(1), throws);
```

Comment:
One Q: Why even allow this error to happen? Is `specifiedType` required in all cases, or are there times where it isn't required? 

As a consumer, if it's always required, I feel like it would make more sense to avoid runtime errors and simply make `specifiedType` a second, non-optional `specifiedType` param. If you still want a check, you could throw in an `assert(specifiedType != null)`.

Or did I totally miss the point of this one?